flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/functions/aggfunctions/MinWithRetractAggFunction.java

```diff
@@ -367,13 +368,21 @@ public void retract(MinWithRetractAccumulator<BinaryString> acc, BinaryString va
 	/**
 	 * Built-in Timestamp Min with retraction aggregate function.
 	 */
-	public static class TimestampMinWithRetractAggFunction extends MinWithRetractAggFunction<Timestamp> {
+	public static class TimestampMinWithRetractAggFunction extends MinWithRetractAggFunction<SqlTimestamp> {
 
 		private static final long serialVersionUID = -7494198823345305907L;
 
+		public void accumulate(MinWithRetractAccumulator<SqlTimestamp> acc, SqlTimestamp value) throws Exception {
+			super.accumulate(acc, value);
+		}
+
+		public void retract(MinWithRetractAccumulator<SqlTimestamp> acc, SqlTimestamp value) throws Exception {
+			super.retract(acc, value);
+		}
+
 		@Override
-		protected TypeInformation<Timestamp> getValueTypeInfo() {
-			return Types.SQL_TIMESTAMP;
+		protected TypeInformation<SqlTimestamp> getValueTypeInfo() {
+			return new SqlTimestampTypeInfo(3);
```

Comment:
I am a little worried about timestamp with precision 9.
Can you verify it?
Maybe we should pass precision to this class.

Reply:
Good point.

Reply:
Actually the TimestampMaxWithRetractAggFunction/TimestampMinWithRetractAggFunction are not support high precision(larger than 3) now. We check this in `AggFunctionFactory`.
But let them support high precision in this PR is OK to me.